packages/fxa-settings/src/lib/hooks/useRedirect/index.tsx

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { isAllowed } from 'fxa-shared/configuration/convict-format-allow-list';
+import {
+  BaseIntegrationData,
+  IntegrationType,
+  isWebIntegration,
+  useConfig,
+  useFtlMsgResolver,
+} from '../../../models';
+import { useLocation } from '@reach/router';
+import {
+  AuthUiErrors,
+  composeAuthUiErrorTranslationId,
+} from '../../auth-errors/auth-errors';
+
+export interface UseRedirectIntegration {
+  type: IntegrationType;
+  data: {
+    redirectTo: BaseIntegrationData['redirectTo'];
+  };
+}
+
+/*
+ * Check if the integration contains a valid `redirectTo` based on
+ * a whitelist maintained in the config.
+ *
+ * At the time of writing, this is only valid for web integrations.
+ * OAuth integrations must derive keys before redirecting.
+ */
+export function useRedirect(integration: UseRedirectIntegration) {
+  const config = useConfig();
+  const location = useLocation();
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const isValid =
+    isWebIntegration(integration) && integration.data.redirectTo
```

Comment:
Why do we care that it's a web intergration?

Reply:
This is to make it painfully obvious that this is not to be used in the oauth flow, just when there's a `redirectTo` for the web integration flow. It's probably not necessary though, I think I'll take it out and instead name this `useWebRedirect`.